### ISSUE TITLE
Remove process manager for salt-minion

### ISSUE THREAD

**Issue by jacksontj** [saltstack/salt#27146]
Discussion from previous issue (#22313), paste of the issue to solve:

```
IMO this "keepalive minion" feature using a separate process is a bit of a hack and we shouldn't have it. If all you want to do is have a minion process always running there are many options out there today (runit, init, systemd, upstart, sysvinit, nosh, finit, watchdogd, etc.) so there is no reason for us to have the same code in salt-- especially when our implementation is very platform specific (*nix only).

I propose we just have the minion re-try on SaltClientError (the only exception tune_in can raise) using the acceptance_wait_time-- but do nothing in the case where the minion is killed (kill -9 or anything similar).
```

**TLDR; there are so many process managers, if all we are doing is restarting the salt minion-- lets use init, runit, or any of the million other variants**

Some other related issues:
- #22342
- #22067
- #14236

**Comment by steverweber** [comment]
https://github.com/saltstack/salt/pull/22313

**Comment by jfindlay** [comment]
ping @cachedout

**Comment by cachedout** [comment]
Yup, I'm aware of this. Thanks! :]
